lib/services/profiles-api-service.js

```diff
@@ -102,41 +102,80 @@ function profileApiServiceFactory(
 
                 });
             } else {
-                return self.createNodeAndRunDiscovery(macAddresses);
+                return self.createNodeAndRunDiscovery(macAddresses, options);
             }
         });
     };
 
-    ProfileApiService.prototype.runDiscovery = function(node) {
+    ProfileApiService.prototype.runDiscovery = function(node, options) {
         var self = this;
+        var configuration;
 
+        if (node.type === 'switch') {
+            configuration = self.getSwitchDiscoveryConfiguration(node, options.switchVendor);
+        } else {
+            configuration = {
+                name: 'Graph.SKU.Discovery',
+                options: {
+                    defaults: {
+                        graphOptions: {
+                            target: node.id
+                        },
+                        nodeId: node.id
+                    }
+                }
+            };
+        }
+
+        // The nested workflow holds the lock against the nodeId in this case,
+        // so don't add it as a target to the outer workflow context
+        return workflowApiService.createAndRunGraph(configuration, null)
+        .then(function() {
+            return self.waitForDiscoveryStart(node.id);
+        })
+        .then(function() {
+            return node;
+        });
+    };
+
+    ProfileApiService.prototype.getSwitchDiscoveryConfiguration = function(node, vendor) {
         var configuration = {
-            name: 'Graph.SKU.Discovery',
+            name: 'Graph.SKU.Switch.Discovery.Active',
             options: {
                 defaults: {
                     graphOptions: {
                         target: node.id
                     },
                     nodeId: node.id
+                },
+                'vendor-discovery-graph': {
+                    graphName: null
                 }
             }
         };
 
-        return workflowApiService.createAndRunGraph(configuration)
-        .then(function() {
-            return self.waitForDiscoveryStart(node.id);
-        })
-        .then(function() {
-            return node;
-        });
+        vendor = vendor.toLowerCase();
+
+        if (vendor === 'cisco') {
```

Comment:
@jlongever yeah I think this is the weakest part of the PR right here, though I also figured we might want to start with something really basic and then evolve the abstraction as our needs grow? 

We do sort vendor type based on mac prefix right now, it's just done in the dhcp configuration file rather than here (since we have to check the mac at that point anyways to know whether it's a switch or a compute node).

The overall workflow that's run **is** a generic one, while there is a nested cataloging workflow that is run as a vendor specific one. Another way to do this that I thought about would be to run a completely generic cataloging workflow, and enforce being vendor agnostic in the python scripts, though I think that has the potential to get pretty messy very quickly.

But as far as a better place for putting vendor type mappings, I haven't come up with a solution yet.

Thoughts?